File: field/9L4/genLoads.py
"""
genLoads.py

Generate ARF excitation field for different focal depths and attenuations.

Mark Palmeri
mlp6@duke.edu
2015-04-05
"""

import os
import shutil

# define some stuff
Freq = [6.13]
Fnum = [1.5]
FD = [0.030]
alpha = [0.5]

root = '/crabtree/mlp6/cervix/field/9L4'

SGE_TEMPLATE = 'genLoadsSGE.py'

for i in range(len(Freq)):
    for j in range(len(Fnum)):
        for k in range(len(FD)):
            for l in range(len(alpha)):
                datafile = '%s/dyna-I-f%.2f-F%.1f-FD%.3f-a%.2f.mat' % \
                           (root, Freq[i], Fnum[j], FD[k], alpha[l])
                if not os.path.exists(datafile):
                    SGE_FILENAME = datafile.replace('dyna-I-', 'field_')
                    SGE_FILENAME = SGE_FILENAME.replace('.mat', '_SGE.py')
                    print SGE_FILENAME
                    shutil.copy(SGE_TEMPLATE, SGE_FILENAME)
                    PARAM_STRING = '%.1f, %.3f, %.2f, %.2f' % \
                                   (Fnum[j], FD[k], Freq[i], alpha[l])
                    os.system('sed -i -e "s/PARAM_STRING/%s/" %s' %
                              (PARAM_STRING, SGE_FILENAME))
                    os.system('qsub --python %s' % (SGE_FILENAME))
                else:
                    print('%s ALREADY EXISTS!!' % datafile)

File: field/9L4/genLoads.m
function genLoads(FNUM, FOCDEPTH, FREQ, ALPHA)
% function genLoads(FNUM, FOCDEPTH, FREQ, ALPHA)
%
% Call FEM/field Field II functions to generate ARF loads.
%
% Mark Palmeri
% mlp6@duke.edu
% 2015-04-05

FDmm = FOCDEPTH * 1000;

nodeFile = sprintf('../../mesh/nodes.dyn', FDmm);

inputIntensityFile = field2dyna(nodeFile, ALPHA, FNUM, [0 0 FOCDEPTH], FREQ, ...
                                'l94', 'gaussian');

normIntensityFile = 'dyna-I-f6.15-F1.5-FD0.050-a0.50.mat';

makeLoadsTemps(inputIntensityFile, normIntensityFile, 1000, 500, 2.0, 0.01^3, 'q', 1);

% generate VTU files for paraview
system(sprintf('python ~/git/fem/mesh/create_pointloads_vtk.py --loadfile PointLoads-f6.15-F%.1f-FD0.0%i-a0.50.dyn --loadout nodeLoadsFoc%immF%.1f', FNUM, FDmm, FDmm, FNUM));

quit
